src/util/message.h

```diff
@@ -44,10 +45,47 @@ struct messaget
     template <typename... Args>
     static void println(FILE *f, VerbosityLevel lvl, Args &&...args)
     {
-      if(lvl == VerbosityLevel::Error)
-        fmt::print(f, "ERROR: ");
-      fmt::print(f, std::forward<Args>(args)...);
-      fmt::print(f, "\n");
+    	     
+      	/** 
+		* Newly added code to address issue #886.
+			* It prepends a well formatted timestamp, file path, function name, line number and log level to each LOG message. 
+			* It is transparent to the existing logging function, i.e., log_message.
+			* TODO: It can be implemented using a separate function and invoked here and/or elsewhere when required. 		
+		*/		
+		//========Start===============================================================================================
+		struct tm* time_struct; //A time structure consisting of 9 members, such as tm_sec, tm_min, tm_hour, tm_mday, tm_mon, tm_year, etc. 
+		char time_info[20]; //A variable to hold the date and time values.
+		const char *log_level;//[12]; //A variable to hold the log level.
+		
+		time_t current_time = time(0);
+		time_struct = localtime(&current_time);
+		strftime (time_info, sizeof(time_info), "%Y-%m-%d %H:%M:%S", time_struct); //"strftime" is to format the date/time structure. 
+	   
+		switch(lvl){//To prepend the respective log level to each log message.  
+			case VerbosityLevel::Error:
+				log_level = "[ERROR]:";
+				break;
+			case VerbosityLevel::Result:				
+				log_level = "[RESULT]:";
+				break;
+			case VerbosityLevel::Warning:				
+				log_level = "[WARNING]:";
+				break;
+			case VerbosityLevel::Progress:				
+				log_level = "[PROGRESS]:";
+				break;
+			case VerbosityLevel::Status:				
+				log_level = "[STATUS]:";
+				break;
+			default:
+				log_level = "[DEBUG]:";
+		}
+		
+		std::fprintf(f, "%s [%s] [%s:%d] %s ", time_info, __FILE__, __FUNCTION__, __LINE__, log_level); /* Prepends the timestamp, file name, file path, 
```

Comment:
Unless I'm mistaken, the expansion of the macros `__FILE__`, `__FUNCTION__` and `__LINE__` here will always default to `message.h`, `println()` and `84`. That's not helpful. With C++20 we could've used [std::source_location](https://en.cppreference.com/w/cpp/utility/source_location), but as ESBMC is C++17, I don't think the only choice we have is to make the log_(error|warning|...) things preprocessor macros instead of functions below, which insert those macros appopriately in a call to `println()`. Alternatively, drop this source-location info from the message.

As an additional comment (more a feature request): would it be of interest to disable this meta data via an envvar or cmd-line flag? I would like to do that in order to remove clutter from the terminal window when debugging issues.